tests/hash-backend-integration/src/tests/subgraph/friendship.test.ts

```diff
@@ -183,6 +188,68 @@ afterAll(async () => {
   await resetGraph();
 });
 
+describe("Ontology queries", () => {
+  it("read data types", async () => {
+    const subgraph = await getDataTypes(graphContext, {
+      query: {
+        filter: {
+          all: [],
+        },
+        graphResolveDepths: {
+          ...zeroedGraphResolveDepths,
+          inheritsFrom: { outgoing: 1 },
```

Comment:
Is there a reason we're doing this rather than leaving them all zeroed?

Reply:
Described in the OP: The error only happens if at least one resolve depth is not zero (this is why this case wasn't covered before)

Reply:
https://github.com/hashintel/hash/pull/2629 short circuited the traversal if no traversal is needed.

Reply:
Ah of course, sorry. I think this is somewhat of a confusing test at the moment then, and it'd be quite easy to accidentally remove our coverage of this if we go and implement inheritance for data types and update the test.

Perhaps we should rewrite the test as one that checks that the number of roots of the subgraph is the same regardless of the depths, so one with `0` depths and one with `255`? (And maybe we pick something that does have edges, such as property types)

Reply:
I added a wider range of resolve depths: [`27bf7af` (#2641)](https://github.com/hashintel/hash/pull/2641/commits/27bf7afb2edbc1b7b477f4a64dc3efdde402baff)

Reply:
One final comment https://github.com/hashintel/hash/pull/2641#discussion_r1218154126

As suggested I think `0` and `255` would be better as it checks our short circuit logic too

Reply:
^ got confused by the syntax, disregard, thanks for clarifying :)